golem-worker-executor-base/src/durable_host/golem/mod.rs

```diff
@@ -462,6 +463,26 @@ impl<Ctx: WorkerCtx> golem::api::host::Host for DurableWorkerCtx<Ctx> {
 
         Ok(())
     }
+
+    async fn get_self_metadata(&mut self) -> anyhow::Result<golem::api::host::WorkerMetadata> {
+        record_host_function_call("golem::api", "get_self_metadata");
+        let metadata = self.state.worker_service.get(&self.owned_worker_id).await;
```

Comment:
This should be something like `self.public_state.worker().get_metadata()` which will return the latest state in-memory instead of querying the last persisted one from redis